### PR TITLE
[feature] add support for EAFT loss

### PR DESCRIPTION
**Pull request by ymxyll**
# PR type
- [ ] Bug Fix
- [✅ ] New Feature
- [ ] Document Updates
- [ ] More Models or Datasets Support

# PR information

Dear all,
Thanks for maintaining such a fantastic project that enables the community to explore various fine-tuning approaches on foundation models.

I would like to contribute by implementing [Entropy-Adaptive Fine-Tuning(EAFT)](https://arxiv.org/abs/2601.02151) support to this repository.

Thank you so much, and I welcome any feedback for improving this PR.

Best regards,

### CODE REVIEW

File: swift/megatron/trainers/trainer.py [hunk 1]
```diff
@@ -52,13 +52,41 @@ def loss_func(self,
...
+                probs_topk = torch.exp(log_probs_topk)
+                entropy_approx = -(probs_topk * log_probs_topk).sum(dim=-1)
+                normalized_entropy = entropy_approx / 3.0
+                eaft_weight_valid = torch.pow(normalized_entropy, args.eaft_alpha)
+                
+                eaft_weight = torch.ones(batch_size * seq_length, device=logits_float.device)
+                eaft_weight[loss_mask.view(-1)] = eaft_weight_valid
+                eaft_weight = eaft_weight.view(batch_size, seq_length)
+
+            losses = losses * eaft_weight
```
[the remark this answers is not in the document]
Comment (ymxyll) [reply swift/megatron/trainers/trainer.py:89]: Thanks for the review. I’ve double-checked the shapes, and the implementation is actually correct. Since the input logits are [Seq, Batch], I explicitly transpose them via transpose(0, 1) to align with the [Batch, Seq] layout of the losses tensor. Given that eaft_weight serves as a weight map for the loss, applying element-wise multiplication here is the correct operation, so no further transposition is required.
